src/makeInfinityAdder.js

```diff
@@ -6,7 +6,23 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  const adder = (num) => {
+    if (num !== undefined) {
+      sum += num;
+
+      return adder;
+    } else {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    }
```

Comment:
`else` is redundant since you've used early `return`
```suggestion
    }
    
    const result = sum;

    sum = 0;

    return result;
```